{
  "wrapper": "plaintext",
  "text": "Hi Bill,\n\nThe exception is from executor side. From the gist you provided, looks like\nthe issue is that you only configured java options in driver side, I think\nyou should also configure this in executor side. You could refer to here (\nhttps://github.com/hortonworks-spark/skc#running-on-a-kerberos-enabled-cluster\n).\n\n    --files key.conf#key.conf,v.keytab#v.keytab\n    --driver-java-options \"-Djava.security.auth.login.config=./key.conf\"\n    --conf \"spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./key.conf\"\n\n\nOn Fri, Mar 31, 2017 at 1:58 AM, Bill Schwanitz <bilsch@bilsch.org> wrote:\n\n> I'm working on a poc spark job to pull data from a kafka topic with\n> kerberos enabled ( required ) brokers.\n>\n> The code seems to connect to kafka and enter a polling mode. When I toss\n> something onto the topic I get an exception which I just can't seem to\n> figure out. Any ideas?\n>\n> I have a full gist up at https://gist.github.com/bilsch/\n> 17f4a4c4303ed3e004e2234a5904f0de with a lot of details. If I use the\n> hdfs/spark client code for just normal operations everything works fine but\n> for some reason the streaming code is having issues. I have verified the\n> KafkaClient object is in the jaas config. The keytab is good etc.\n>\n> Guessing I'm doing something wrong I just have not figured out what yet!\n> Any thoughts?\n>\n> The exception:\n>\n> 17/03/30 12:54:00 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0,\n> host5.some.org.net): org.apache.kafka.common.KafkaException: Failed to\n> construct kafka consumer\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:702)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:557)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:540)\n> at org.apache.spark.streaming.kafka010.CachedKafkaConsumer.<\n> init>(CachedKafkaConsumer.scala:47)\n> at org.apache.spark.streaming.kafka010.CachedKafkaConsumer$.\n> get(CachedKafkaConsumer.scala:157)\n> at org.apache.spark.streaming.kafka010.KafkaRDD$KafkaRDDIterator.<init>(\n> KafkaRDD.scala:210)\n> at org.apache.spark.streaming.kafka010.KafkaRDD.compute(\n> KafkaRDD.scala:185)\n> at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:319)\n> at org.apache.spark.rdd.RDD.iterator(RDD.scala:283)\n> at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:70)\n> at org.apache.spark.scheduler.Task.run(Task.scala:86)\n> at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)\n> at java.util.concurrent.ThreadPoolExecutor.runWorker(\n> ThreadPoolExecutor.java:1142)\n> at java.util.concurrent.ThreadPoolExecutor$Worker.run(\n> ThreadPoolExecutor.java:617)\n> at java.lang.Thread.run(Thread.java:745)\n> Caused by: org.apache.kafka.common.KafkaException:\n> org.apache.kafka.common.KafkaException: Jaas configuration not found\n> at org.apache.kafka.common.network.SaslChannelBuilder.\n> configure(SaslChannelBuilder.java:86)\n> at org.apache.kafka.common.network.ChannelBuilders.\n> create(ChannelBuilders.java:70)\n> at org.apache.kafka.clients.ClientUtils.createChannelBuilder(\n> ClientUtils.java:83)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:623)\n> ... 14 more\n> Caused by: org.apache.kafka.common.KafkaException: Jaas configuration not\n> found\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.getServiceName(\n> KerberosLogin.java:299)\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.configure(\n> KerberosLogin.java:103)\n> at org.apache.kafka.common.security.authenticator.LoginManager.<init>(\n> LoginManager.java:45)\n> at org.apache.kafka.common.security.authenticator.LoginManager.\n> acquireLoginManager(LoginManager.java:68)\n> at org.apache.kafka.common.network.SaslChannelBuilder.\n> configure(SaslChannelBuilder.java:78)\n> ... 17 more\n> Caused by: java.io.IOException: Could not find a 'KafkaClient' entry in\n> this configuration.\n> at org.apache.kafka.common.security.JaasUtils.jaasConfig(\n> JaasUtils.java:50)\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.getServiceName(\n> KerberosLogin.java:297)\n> ... 21 more\n>\n\n",
  "denotations": [
    {
      "id": 1,
      "start": 0,
      "end": 9,
      "text": "Hi Bill,\n",
      "type": "Body/Intro",
      "meta": null
    },
    {
      "id": 2,
      "start": 0,
      "end": 532,
      "text": "Hi Bill,\n\nThe exception is from executor side. From the gist you provided, looks like\nthe issue is that you only configured java options in driver side, I think\nyou should also configure this in executor side. You could refer to here (\nhttps://github.com/hortonworks-spark/skc#running-on-a-kerberos-enabled-cluster\n).\n\n    --files key.conf#key.conf,v.keytab#v.keytab\n    --driver-java-options \"-Djava.security.auth.login.config=./key.conf\"\n    --conf \"spark.executor.extraJavaOptions=-Djava.security.auth.login.config=./key.conf\"\n\n\n",
      "type": "Body",
      "meta": null
    },
    {
      "id": 3,
      "start": 532,
      "end": 607,
      "text": "On Fri, Mar 31, 2017 at 1:58 AM, Bill Schwanitz <bilsch@bilsch.org> wrote:\n",
      "type": "Header",
      "meta": null
    },
    {
      "id": 4,
      "start": 607,
      "end": 4119,
      "text": "\n> I'm working on a poc spark job to pull data from a kafka topic with\n> kerberos enabled ( required ) brokers.\n>\n> The code seems to connect to kafka and enter a polling mode. When I toss\n> something onto the topic I get an exception which I just can't seem to\n> figure out. Any ideas?\n>\n> I have a full gist up at https://gist.github.com/bilsch/\n> 17f4a4c4303ed3e004e2234a5904f0de with a lot of details. If I use the\n> hdfs/spark client code for just normal operations everything works fine but\n> for some reason the streaming code is having issues. I have verified the\n> KafkaClient object is in the jaas config. The keytab is good etc.\n>\n> Guessing I'm doing something wrong I just have not figured out what yet!\n> Any thoughts?\n>\n> The exception:\n>\n> 17/03/30 12:54:00 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0,\n> host5.some.org.net): org.apache.kafka.common.KafkaException: Failed to\n> construct kafka consumer\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:702)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:557)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:540)\n> at org.apache.spark.streaming.kafka010.CachedKafkaConsumer.<\n> init>(CachedKafkaConsumer.scala:47)\n> at org.apache.spark.streaming.kafka010.CachedKafkaConsumer$.\n> get(CachedKafkaConsumer.scala:157)\n> at org.apache.spark.streaming.kafka010.KafkaRDD$KafkaRDDIterator.<init>(\n> KafkaRDD.scala:210)\n> at org.apache.spark.streaming.kafka010.KafkaRDD.compute(\n> KafkaRDD.scala:185)\n> at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:319)\n> at org.apache.spark.rdd.RDD.iterator(RDD.scala:283)\n> at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:70)\n> at org.apache.spark.scheduler.Task.run(Task.scala:86)\n> at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:274)\n> at java.util.concurrent.ThreadPoolExecutor.runWorker(\n> ThreadPoolExecutor.java:1142)\n> at java.util.concurrent.ThreadPoolExecutor$Worker.run(\n> ThreadPoolExecutor.java:617)\n> at java.lang.Thread.run(Thread.java:745)\n> Caused by: org.apache.kafka.common.KafkaException:\n> org.apache.kafka.common.KafkaException: Jaas configuration not found\n> at org.apache.kafka.common.network.SaslChannelBuilder.\n> configure(SaslChannelBuilder.java:86)\n> at org.apache.kafka.common.network.ChannelBuilders.\n> create(ChannelBuilders.java:70)\n> at org.apache.kafka.clients.ClientUtils.createChannelBuilder(\n> ClientUtils.java:83)\n> at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(\n> KafkaConsumer.java:623)\n> ... 14 more\n> Caused by: org.apache.kafka.common.KafkaException: Jaas configuration not\n> found\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.getServiceName(\n> KerberosLogin.java:299)\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.configure(\n> KerberosLogin.java:103)\n> at org.apache.kafka.common.security.authenticator.LoginManager.<init>(\n> LoginManager.java:45)\n> at org.apache.kafka.common.security.authenticator.LoginManager.\n> acquireLoginManager(LoginManager.java:68)\n> at org.apache.kafka.common.network.SaslChannelBuilder.\n> configure(SaslChannelBuilder.java:78)\n> ... 17 more\n> Caused by: java.io.IOException: Could not find a 'KafkaClient' entry in\n> this configuration.\n> at org.apache.kafka.common.security.JaasUtils.jaasConfig(\n> JaasUtils.java:50)\n> at org.apache.kafka.common.security.kerberos.KerberosLogin.getServiceName(\n> KerberosLogin.java:297)\n> ... 21 more\n>\n\n",
      "type": "Body",
      "meta": null
    }
  ],
  "meta": {},
  "id": "test/train_2232"
}